### PR TITLE
POC for scheduled API calls to import-service.

### PR DESCRIPTION
**Pull request by bgsavage**
This is just a prototype, I don't plan to merge this at at this time, but looking for feedback on my approach to use Eureka as a means for client discovery and facilitation of cross service communication.  I'm not sure if I need Feign in there at this time, I'm still testing it out.  

Added a new task service which holds a task scheduler.  This also doubles as a Eureka server to enable cross service communication via feign.

Right now just hitting a test endpoint in the import-service

### CODE REVIEW

File: import-service/build.gradle [hunk 1]
```diff
@@ -44,6 +44,10 @@ dependencies {
     compile 'org.opentestsystem.rdw.common:rdw-common-utils'
     compile 'mysql:mysql-connector-java'
 
+    // enables to client discovery through Eureka
```
Comment (chdietz) [inline import-service/build.gradle]: your thinking about eureka for angular client to discover API server or internal service to find other services?  This is new to me sounds like it is not for edge services.

Comment (chdietz) [inline import-service/build.gradle]: ok, I see further down the discoveryClient

Comment (chdietz) [inline import-service/build.gradle]: @malaffoon does k8s provide discovery similar to this?

File: task-service/build.gradle [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+apply plugin: 'org.springframework.boot'
+
+dependencies {
+    // handy for creating metadata with @ConfigurationProperties
+    optional "org.springframework.boot:spring-boot-configuration-processor"
+
+    compile 'org.springframework.boot:spring-boot-starter-actuator'
+
+    compile 'org.springframework.cloud:spring-cloud-starter-config'
+    compile 'org.springframework.cloud:spring-cloud-stream'
```
Comment (malaffoon) [inline task-service/build.gradle]: doubt you need cloud-stream

Comment (bgsavage) [inline task-service/build.gradle]: probably not, I just used another project in ingest to form a gradle build template and removed what I know I didn't need.

**Comment by malaffoon** [comment]
I don't see a huge value-add with Eureka here. Near as i can tell all it is doing is making `rdw-ingest-import-service` resolve to something. But, in an orchestrated environment (either docker-compose or k8s) that is something that can be made available. For example, look at the configuration service or rabbitmq. We could either have a property (e.g. `import-service.host`) or even "hard-code" a well-known alias for the host knowing that orchestration can make it available.

And the one thing i was hoping for isn't being provided: a wrapped rest template that has methods mapped to end-points. So, instead of specifying "http://rdw-ingest-import-service/exams/imports/reconciliation" with params and all that, you would just use something like `template.getReconciliation(...)`.

**Comment by bgsavage** [comment]
The repository layer is serving as my wrapped template here.  Does this work for you?

**Comment by malaffoon** [comment]
I understand that we can abstract the functionality behind a repository, and what you've done is exactly that and good. But i was hoping that you wouldn't need to have any (visible) http code.

**Comment by bgsavage** [comment]
Oh similar to how we used to generate code from a wsdl for soap?

I'll go ahead and remove Eureka.  I just thought it was really cool and easy to stand up, but I can just use configured endpoints just as easy if this is preferred.  I guess I had a different understanding what we were trying to do here.

Thanks for the feedback!

**Pull request closed without merging**